docs/source/executing-operations/defer-support.mdx

```diff
@@ -0,0 +1,170 @@
+
+Here's an example query that uses `@defer`:
+
+```graphql
+query GetPersonAndFriends($personId: ID!) {
+  person(id: $personId) {
+    # Basic fields (fast)
+    id
+    name {
+      firstName
+      lastName
+    }
+
+    # highlight-start
+    # Friend fields (slower)
+    ... on User @defer {
+      friends {
+        id
+      }
+    }
+    # highlight-end
+  }
+}
+```
+
+When responding to the above query, a GraphQL server can "chunk" its response, starting by returning only the "basic fields". Later, it can return an additional chunk that includes the "friend fields".
+
+As shown, deferred fields are always contained within a GraphQL fragment, and the `@defer` directive is applied to that fragment (_not_ to the individual fields).
+
+## Using `@defer` with the router
+
+> To use `@defer` successfully with the Apollo Router, a client's GraphQL library must _also_ support the directive by handling "chunked" responses correctly.
+
+The Apollo Router supports deferring the following fields in your schema:
+
+* Fields that are nested anywhere within an [entity type](#entity-fields)
+* Root fields of the [`Query` type](#fields-of-query)
+
+In most supergraphs, this represents a large percentage of all fields in the schema. [See an example of unsupported fields.](#unsupported-field-example)
+
+### Entity fields
+
+The Apollo Router supports using `@defer` with fields and nested subfields of [entity types](/federation/entities/) in your federated supergraph. Entities are object types with at least one `@key`, which specifies the fields that can uniquely identify any given instance of that type.
+
+Here's an example subgraph schema that defines a `User` entity:
+
+```graphql title="Users subgraph"
+type User @key(fields: "id") {
+  id: ID!
+  name: Name!
+  friends: [User!]!
+}
+
+type Name {
+  firstName: String!
+  lastName: String!
+}
+
+type Query {
+  person(id: ID!): User
+}
+```
+
+Because the `User` type is an entity, a client query can apply `@defer` to any fragment that's nested somewhere within that type (at any depth).
+
+Here are two example queries:
+
+```graphql
+query QueryA($personId: ID!) {
+  person(id: $personId) {
+    id
+    # highlight-start
+    ... on User @defer {
+      name {
+        firstName
+        lastName
+      }
+    }
+    # highlight-end
+  }
+}
+
+query QueryB($personId: ID!) {
+  person(id: $personId) {
+    id
+    name {
+      firstName
+      # highlight-start
+      ... on Name @defer {
+        lastName
+      }
+      # highlight-end
+    }
+  }
+}
+```
+
+### Fields of `Query`
+
+The Apollo Router also supports using `@defer` with root fields of your supergraph's `Query` type:
+
+```graphql
+query PartiallyDeferredQuery {
+  products {
+    id
+  }
+  # highlight-start
+  ... on Query @defer {
+    users {
+      id
+    }
+  }
+  # highlight-end
+}
+```
+
+You _cannot_ use `@defer` with root fields of the `Mutation` type.
+
+### Unsupported field example
```

Comment:
The example in that section is generally correct, and nothing said here is exactly false, but I worried that because it only consider a fairly simple case, it migth leave users wondering what to expect in more complex/realistic example (or even maybe making incorrect generalisation).

Let me illustrate with the following schema (using a single-subgraph for simplicity; it's irrelevant to my points):
```graphql
type Query {
  gimmeUser: User
}

type User @key(fields: "id") {
  id: ID!
  name: String
  taggedVideos: TaggedVideo[]
}

type TaggedVideo {
  tag: string
  video: Video
}

scalar Bytes

type Video @key(fields: "id") {
  id: ID!
  name: String
  data: Bytes
  comments: Comment[]
}

type Comment {
  author: User
  text: String
}

```
and query:
```graphql
{
  gimmeUser {
    name
    taggedVideos {
      tag
      ... @defer {
        video {
          data
          comments {
            author {
              name
            }
          }
        }
      }
    }
  }
}
```
Now, the first deferred field is `TaggedVideo.video` and that is not an entity field, so we cannot actually defer its resolution. But that does not mean _nothing_ gets deferred either.
Instead, the query plan will essentially send:
```graphql
{
  gimmeUser {
    name
    taggedVideos {
      tag
      video {
        id
      }
    }
  }
  }
}
```
and then will be able to return the "primary" (non-deferred) chunk. That first chunk won't have any of the requested `TaggedVideo.video` stuff, though being the hood we will already have resolved the ID of all the corresponding videos.

Then the 2nd request made to the subgraph internally will be:
```graphql
{
  _entities(...) {
    ... on Video {
      data
      comments {
        author {
          name
        }
      }
    }
  }
}
```
and that will feed the 2nd response chunk, the deferred one.

The point here is that it's not because a `@defer` starts on a non-entity field that nothing gets deferred. And in general, to known what gets deferred or not, you want to look at fields at the root of the `@defer` ( and if those aren't deferred, look at the ones within those etc...).

Reply:
Thanks @pcmanus! I've overhauled the examples and clarified at the top of the "Non-deferred fields" section how the router gracefully handles partial defer-ability of a fragment. Any followup comments welcome 🙏